homeassistant/components/axion_dmx/light.py

```diff
@@ -0,0 +1,337 @@
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Instruct the light to turn off."""
+        _LOGGER.debug(f"Instructing the {self._name} to turn off!")
+        self._is_on = False
+        await self.api.set_level(self._channel, 0)
+
+        if self._light_type in ["RGB", "RGBW", "RGBWW"]:
+            self._last_hs_color = self._attr_hs_color
+            await self.api.set_level((self._channel + 1), 0)
+            await self.api.set_level((self._channel + 2), 0)
+            if self._light_type in ["RGBW", "RGBWW"]:
+                self._last_rgbw = self._attr_rgbw_color
+                await self.api.set_level((self._channel + 3), 0)
+                if self._light_type == "RGBWW":
+                    self._last_rgbww = self._attr_rgbww_color
+                    await self.api.set_level((self._channel + 4), 0)
+        elif self._light_type == "Tunable White":
+            await self.api.set_level(self._channel + 1, 0)
+
+        # Add a small delay to allow the controller to process the command
+        await asyncio.sleep(0.5)
+
+        # Manually refresh the coordinator to get the latest state
+        await self.coordinator.async_request_refresh()
+        self.async_write_ha_state()
+
+    async def async_update(self) -> None:
+        """Fetch new state data for this light."""
+        level = await self.api.get_level(self._channel)
+        self._is_on = level > 0
+        self._attr_brightness = level
+
+        if self._light_type in ["RGB", "RGBW", "RGBWW"]:
+            # Assuming the RGB values are stored in a way to fetch them
+            r = await self.api.get_level(self._channel)
+            g = await self.api.get_level(self._channel + 1)
+            b = await self.api.get_level(self._channel + 2)
+            hs_color = color_util.color_RGB_to_hs(r, g, b)
+            self._attr_hs_color = (int(hs_color[0]), int(hs_color[1]))
+
+            # Determine if the light is on based on any of the RGB channels being non-zero
+            self._is_on = any([r, g, b])
+            self._attr_brightness = max(r, g, b)
+
+            if self._light_type in ["RGBW", "RGBWW"]:
+                w1 = await self.api.get_level(self._channel + 3)
+                self._attr_rgbw_color = (r, g, b, w1)
+                self._is_on = any([r, g, b, w1])
+                self._attr_brightness = max(r, g, b, w1)
+
+                if self._light_type == "RGBWW":
+                    w2 = await self.api.get_level(self._channel + 4)
+                    self._attr_rgbww_color = (r, g, b, w1, w2)
+                    self._is_on = any([r, g, b, w1, w2])
+                    self._attr_brightness = max(r, g, b, w1, w2)
+
+        elif self._light_type == "Tunable White":
+            cold_white_level = await self.api.get_level(self._channel)
+            warm_white_level = await self.api.get_level(self._channel + 1)
+            self._is_on = any([cold_white_level, warm_white_level])
```

Comment:
We have a coordinator, what do we use this for?

Reply:
This is cool, didn't knew it'll work this good. Thanks for the heads up!